tests/frontend/cli/test_starship.py

```diff
@@ -0,0 +1,506 @@
+        """Tests that if a token was given as a command line argument then
+        configuration takes place accordingly."""
+        with monkeypatch.context() as m:
+            mock_store_account = MockStoreAccount()
+            m.setattr(cli, "store_account", mock_store_account.store_account)
+
+            args = MockArgs()
+            args.token = "SomeToken"
+
+            cli.configure(args)
+            assert mock_store_account.kwargs == {"authentication_token": "SomeToken"}
+
+    def test_configure_everything(self, monkeypatch):
+        """Tests that if no token was given as a command line argument then
+        configuration takes place using the configure_everything function."""
+        with monkeypatch.context() as m:
+            mock_store_account = MockStoreAccount()
+            m.setattr(cli, "configure_everything", lambda: cli.create_config()["api"])
+            m.setattr(cli, "store_account", mock_store_account.store_account)
+
+            args = MockArgs()
+            args.token = False
+
+            cli.configure(args)
+            assert mock_store_account.kwargs == EXPECTED_KWARGS
+
+    def test_token_local(self, monkeypatch):
+        """Tests that if a token was given as a command line argument and
+        local configuration was specified then configuration takes place
+        accordingly."""
+        with monkeypatch.context() as m:
+            mock_store_account = MockStoreAccount()
+            m.setattr(cli, "store_account", mock_store_account.store_account)
+
+            args = MockArgs()
+            args.token = "SomeToken"
+            args.local = True
+
+            cli.configure(args)
+            assert mock_store_account.kwargs == {"authentication_token": "SomeToken", "location": "local"}
+
+    def test_configure_everything_local(self, monkeypatch):
+        """Tests that if no token was given as a command line argument and
+        local configuration was specified then configuration takes place using
+        the configure_everything function."""
+        with monkeypatch.context() as m:
+            mock_store_account = MockStoreAccount()
+            m.setattr(cli, "configure_everything", lambda: cli.create_config()["api"])
+            m.setattr(cli, "store_account", mock_store_account.store_account)
+
+            args = MockArgs()
+            args.token = False
+            args.local = True
+
+            cli.configure(args)
+            EXPECTED_KWARGS["location"] = "local"
+
+            assert mock_store_account.kwargs == EXPECTED_KWARGS
+
+class MockConnection:
+    """A mock Connection class used for testing."""
+
+    def __init__(self):
+        self.pinging = None
+
+    def ping(self):
+        self.pinging = "SuccessfulPing"
+
+class MockSysStdout:
+    """A mock class used for mocking the sys.stdout object while testing."""
+
+    def __init__(self):
+        self.write_output = []
+
+    def write(self, message):
+        self.write_output = message
```

Comment:
In general, I would suggest using `capfd` or `capsys` built in fixtures rather than mocking out sys.stdout manually

Reply:
Oh that's a good one, thanks so much! Modified these ones where applicable